src/RoveComm/RoveCommUDP.h

```diff
@@ -0,0 +1,100 @@
+/******************************************************************************
+ * @brief The RoveCommUDP class is used to send and receive data over a UDP
+ *        connection.
+ *
+ * @file RoveCommUDP.h
+ * @author Eli Byrd (edbgkk@mst.edu)
+ * @date 2024-02-06
+ *
+ * @copyright Copyright Mars Rover Design Team 2024 - All Rights Reserved
+ ******************************************************************************/
+
+#ifndef ROVECOMM_UDP_H
+#define ROVECOMM_UDP_H
+
+#include <arpa/inet.h>
+#include <csignal>
+#include <cstring>
+#include <functional>
+#include <iostream>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <unordered_set>
+#include <vector>
+
+#include "../interfaces/AutonomyThread.hpp"
+#include "ExternalIncludes.h"
+#include "RoveCommConsts.h"
+#include "RoveCommGlobals.h"
+#include "RoveCommManifest.h"
+#include "RoveCommPacket.h"
+
+/******************************************************************************
+ * @brief The RoveComm namespace contains all of the functionality for the
+ *        RoveComm library. This includes the packet structure and the
+ *        functions for packing and unpacking data.
+ *
+ * @author Eli Byrd (edbgkk@mst.edu)
+ * @date 2024-02-07
+ ******************************************************************************/
+namespace rovecomm
+{
+    /******************************************************************************
+     * @brief The RoveCommUDP class is used to send and receive data over a UDP
+     *        connection.
+     *
+     * @author Eli Byrd (edbgkk@mst.edu)
+     * @date 2024-02-07
+     ******************************************************************************/
+    class RoveCommUDP : AutonomyThread<void>
+    {
+        private:
+            int nUDPSocket;
```

Comment:
Add some comments or comment blocks to make the private and public sections look cleaner and separate variable declarations from method declarations.